sql/catalyst/src/test/scala/org/apache/spark/sql/RowTest.scala

```diff
@@ -121,10 +121,6 @@ class RowTest extends FunSpec with Matchers {
       externalRow should be theSameInstanceAs externalRow.copy()
     }
 
-    it("copy should return same ref for internal rows") {
```

Comment:
As discussed offline: I don't see the point of having an immutable `GenericInternalRow` if we cannot guarantee its immutability. We could just make every InternalRow a mutable one, and simplify the class structure in the process. I am not sure if we should make that part of the current PR though.